### ISSUE TITLE
Renaming ConnectionTypes isn't working

### ISSUE THREAD

**Issue by cshadek** [GraphQLSwift/Graphiti#106]
Renaming connection types, doesn't seem to fully work.

```
Type(ChatObject.self, as: "Chat") {
    Field("messages", at: ChatObject.messages, as: Connection<MessageObject>.self) {
                    Argument("first", at: \.first)
                    Argument("after", at: \.after)
                    Argument("last", at: \.last)
                    Argument("before", at: \.before)
    }
}

Type(MessageObject.self, as: "Message") {
    Field("id", at: \.id)
    Field("text", at: \.text)
}

ConnectionType(MessageObject.self, as: "Message")
```

**I get the following error:**
_Fatal error: 'try!' expression unexpectedly raised an error: Type "MessageObjectConnection" was referenced but not defined._

When I don't rename the ConnectionType, then the schema builds, but I end up with `MessageObjectConnection` instead of `MessageConnection`.

If I don't include the `Chat` type, then the schema builds with the correct `MessageConnection` name.

**Comment by cshadek** [comment]
@d-exclaimation do you have an idea of how to fix this?

**Comment by d-exclaimation** [comment]
Not 100% sure what's the problem. My guess is probably the fact the Message Object Connection is referenced in Chat type and renaming it somehow breaks that. I haven't used it personally in a while hut I can take a look at it

**Comment by cshadek** [comment]
While on this topic, there's probably some sense in rethinking the Connection API sometime.  

Some nice improvements would be:

- Allowing connections of protocols or unions
- Allowing custom fields on connections
- Allowing custom fields on edges

These improvements would arguably make the Connection feature that much more powerful.

I have a bit of an idea of how this could be done, but my understanding is that there are issues around how to make this work with Codable.

**Comment by cshadek** [comment]
@d-exclaimation the renaming issue is obviously more pressing (though it's still not the end of the world), but I thought you may have some ideas on everything else too

**Comment by cshadek** [comment]
The main issue is fixed as of #109.

**Comment by cshadek** [comment]
Since the main issue is fixed, I created a new issue #110 to discuss other improvements to connections that were mentioned above.
